### PR TITLE
Fix/mqtt-log

### PR DESCRIPTION
**Pull request by samuelbles07**
## Changes

Skip initializing mqtt client when mqtt is not configured on the device

## Result

Screen Shot 2024-09-15 at 22 54 31

### CODE REVIEW

File: examples/OneOpenAir/OneOpenAir.ino [hunk 1]
```diff
@@ -382,7 +382,13 @@ static void createMqttTask(void) {
 }
 
 static void initMqtt(void) {
-  if (mqttClient.begin(configuration.getMqttBrokerUri())) {
+  String mqttUri = configuration.getMqttBrokerUri();
+  if (mqttUri.isEmpty()) {
+    Serial.println("MQTT is not configured, skip initialize mqtt client");
```
Comment (nick-4711) [inline examples/OneOpenAir/OneOpenAir.ino:387]: propose to rephrase to

.., skipping initialization of mqtt client

and decide if we want to use capitalize or not. Maybe 

"MQTT URL  is not configured, , skipping initialization of MQTT client

File: examples/OneOpenAir/OneOpenAir.ino [hunk 2]
```diff
@@ -382,7 +382,13 @@ static void createMqttTask(void) {
...
+    return;
+  }
+
+  if (mqttClient.begin(mqttUri)) {
     Serial.println("Connect to MQTT broker successful");
```
Comment (nick-4711) [inline examples/OneOpenAir/OneOpenAir.ino:392]: Here we could write the past to indicate that it actually worked

"Connected to MQTT broker successfully"

File: examples/OneOpenAir/OneOpenAir.ino [hunk 3]
```diff
@@ -382,6 +382,12 @@ static void createMqttTask(void) {
 }
 
 static void initMqtt(void) {
+  String mqttUri = configuration.getMqttBrokerUri();
+  if (mqttUri.isEmpty()) {
+    Serial.println("MQTT is not configured, skip initialize mqtt client");
```
Comment (pnt325) [inline examples/OneOpenAir/OneOpenAir.ino:387]: There are log message in `mqttClient.begin`. Just update the log message inside is good enough. Cause MQTT also usage on ESP8266 device too. 

https://github.com/samuelbles07/arduino/blob/bf1a55793768ff8240201e70ac822b6bcd6f022d/src/MqttClient.cpp#L20-L28

Comment (samuelbles07) [reply examples/OneOpenAir/OneOpenAir.ino:387]: Yes, but it return false. Hence in the implementation side will print `Connect to MQTT broker failed`. Just realized it is used for others too, will update about this.

Comment (pnt325) [reply examples/OneOpenAir/OneOpenAir.ino:387]: Don't forget to test with ESP8266 too.

**Comment by samuelbles07** [comment]
Screen Shot 2024-09-16 at 15 22 58
Screen Shot 2024-09-16 at 15 29 34

File: examples/BASIC/BASIC.ino [hunk 4]
```diff
@@ -266,10 +266,17 @@ static void mdnsInit(void) {
...
+    Serial.println(
+        "MQTT is not configured, skipping initialization of MQTT client");
+    return;
+  }
+
+  if (mqttClient.begin(mqttUri)) {
+    Serial.println("Successfully connected to MQTT broker");
   } else {
-    Serial.println("setup Connect to MQTT broker failed");
+    Serial.println("Connect to MQTT broker failed");
```
Comment (nick-4711) [inline examples/BASIC/BASIC.ino:279]: Propose to rename to

Connection to MQTT broker failed

File: examples/DiyProIndoorV3_3/DiyProIndoorV3_3.ino [hunk 5]
```diff
@@ -264,10 +264,17 @@ static void mdnsInit(void) {
...
+    Serial.println(
+        "MQTT is not configured, skipping initialization of MQTT client");
+    return;
+  }
+
+  if (mqttClient.begin(mqttUri)) {
+    Serial.println("Successfully connected to MQTT broker");
   } else {
-    Serial.println("setup Connect to MQTT broker failed");
+    Serial.println("Connect to MQTT broker failed");
```
Comment (nick-4711) [inline examples/DiyProIndoorV3_3/DiyProIndoorV3_3.ino:277]: dito

File: examples/BASIC/BASIC.ino (hunk 4 above)
Comment (samuelbles07) [reply examples/BASIC/BASIC.ino]: Updated

**Review by nick-4711 — APPROVED** [review]

**Pull request merged**